rollup.config.js

```diff
@@ -0,0 +1,20 @@
+import typescript from 'rollup-plugin-typescript2';
+import autoExternal from 'rollup-plugin-auto-external';
```

Comment:
What does this do in our case? I'm new to Rollup

Reply:
- `typescript` runs the code through `tsc`
- `autoExternal` sets `dependencies` and `peerDependencies` as external packages so the bundled code leaves them as `import`s. Non-external packages will be bundled in the compiled output.

`rollup-plugin-auto-external` could be replaced with an inlined version. It effectively does the following:

```js
import pkg from './package.json'

// Returns a function that returns true if the provided package name is in `externalArr`.
const makeExternalPredicate = externalArr => {
  if (externalArr.length === 0) {
    return () => false
  }
  const pattern = new RegExp(`^(${externalArr.join('|')})($|/)`)
  return id => pattern.test(id)
}

export default {
  input: 'src/index.ts',
  output: [
    { file: pkg.main, format: 'cjs' },
    { file: pkg.module, format: 'es' },
  ],
  external: makeExternalPredicate([
    ...Object.keys(pkg.dependencies || {}),
    ...Object.keys(pkg.peerDependencies || {}),
  ]),
  plugins: [typescript()],
}
```